### ISSUE TITLE
Writing rules to depend on files generated by ppx

### ISSUE THREAD

**Issue by zbaylin** [ocaml/dune#8981]
[(more context in the OCaml discuss forum)
](https://discuss.ocaml.org/t/generating-files-from-ppx-with-dune)

## Desired Behavior

I have a ppx that consumes some AST node and writes to a file in the CWD. This file can be includes in a rule so that it's copied to the `_build` directory.

## Example

I would like to be able to specify a rule such as
```
(rule
  (deps my_file.pp.ml)
  (targets my_file.txt)
  (action (ppx my_ppx)))
```

The exact syntax isn't set in stone, but ideally I would be able to write a rule like this for every file in a library, etc. that uses a ppx.

**Comment by emillon** [comment]
Hi
- invoking `ppx` as an action has been implemented in #7373
- iterating through all files is going to be difficult since the targets will be different each time, and dune can not have dynamic targets in that sense

**Comment by zbaylin** [comment]
Thanks @emillon, I'll close this issue as it seems that it will be solved by #7373. Iterating through all of the files is a nice-to-have but not required. Thanks again.
